packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -721,189 +519,159 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
-        } else {
-            require(msg.value == 0, "ETH attached for buy auction");
-            // Receiving wSqueeth and paying ETH
-            require(auctionWSqueethEthPrice >= _limitPrice, "Auction price < min price");
-            _executeBuyAuction(msg.sender, wSqueethToAuction, ethProceeds, false);
-        }
+    function nonces(address owner) external view returns (uint256) {
+        return _nonces[owner].current();
+    }
 
-        emit Hedge(
-            msg.sender,
-            _isStrategySellingWSqueeth,
-            _limitPrice,
-            auctionWSqueethEthPrice,
-            wSqueethToAuction,
-            ethProceeds
-        );
+    // solhint-disable-next-line func-name-mixedcase
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return _domainSeparatorV4();
     }
 
-    /**
-     * @notice execute arb between auction price and uniswap price
-     * @param _auctionTriggerTime auction starting time
-     * @param _minWSqueeth minimum WSqueeth amount the caller willing to receive if hedge auction is selling WSqueeth
-     * @param _minEth minimum ETH amount the caller is willing to receive if hedge auction is buying WSqueeth
-     */
-    function _hedgeOnUniswap(
-        uint256 _auctionTriggerTime,
-        uint256 _minWSqueeth,
-        uint256 _minEth
+    function _execOrder(
+        uint256 managerSellAmount,
+        uint256 managerBuyPrice,
+        uint256 sellerPrice,
+        Order memory _order
     ) internal {
-        (
-            bool isSellingAuction,
-            uint256 wSqueethToAuction,
-            uint256 ethProceeds,
-            uint256 auctionWSqueethEthPrice
-        ) = _startAuction(_auctionTriggerTime);
-
-        if (isSellingAuction) {
-            _exactOutFlashSwap(
-                wPowerPerp,
-                weth,
-                IUniswapV3Pool(ethWSqueethPool).fee(),
-                ethProceeds,
-                wSqueethToAuction,
-                uint8(FLASH_SOURCE.FLASH_HEDGE_SELL),
-                abi.encodePacked(wSqueethToAuction, ethProceeds, _minWSqueeth, _minEth)
-            );
-        } else {
-            _exactOutFlashSwap(
-                weth,
-                wPowerPerp,
-                IUniswapV3Pool(ethWSqueethPool).fee(),
-                wSqueethToAuction,
-                ethProceeds,
-                uint8(FLASH_SOURCE.FLASH_HEDGE_BUY),
-                abi.encodePacked(wSqueethToAuction, ethProceeds, _minWSqueeth, _minEth)
-            );
-        }
+        require(managerBuyPrice > 0, "C0");
+        require(managerBuyPrice >= sellerPrice, "C1");
+        bytes32 structHash = keccak256(
+            abi.encode(
+                _CRAB_BALANCE_TYPEHASH,
+                _order.bidId,
+                _order.trader,
+                _order.traderToken,
+                _order.traderAmount,
+                _order.managerToken,
+                _order.managerAmount,
+                _useNonce(_order.trader)
```

Comment:
Is the nonce doing anything like this? If it's to allow order cancellation then shouldn't trader specify nonce and then we check?